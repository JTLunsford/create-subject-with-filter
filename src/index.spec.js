const { createSubject } = require('./index')

describe('subject', () => {
	it('should sub', () => {
		const subject = createSubject()
		subject.subscribe((...args) => expect(args).toMatchSnapshot(), (...args) => expect(args).toMatchSnapshot())
		subject.next('test 1')
		subject.next('test 2')
		subject.next('test 3')
	})
	it('should unsub', () => {
		const subject = createSubject()
		const { unsubscribe } = subject.subscribe((...args) => expect(args).toMatchSnapshot(), (...args) => expect(args).toMatchSnapshot())
		subject.next('test 1')
		unsubscribe()
		subject.next('test 2')
		subject.next('test 3')
	})

	it('should ignore double unsub', () => {
		const subject = createSubject()
		const { unsubscribe } = subject.subscribe((...args) => expect(args).toMatchSnapshot(), (...args) => expect(args).toMatchSnapshot())
		subject.next('test 1')
		unsubscribe()
		unsubscribe()
		subject.next('test 2')
		subject.next('test 3')
	})

	it('should error', () => {
		const subject = createSubject()
		const { unsubscribe } = subject.subscribe((...args) => expect(args).toMatchSnapshot(), (...args) => expect(args).toMatchSnapshot())
		subject.next('test 1')
		subject.error(new Error('error 1'))
		subject.next('test 3')
	})
})
