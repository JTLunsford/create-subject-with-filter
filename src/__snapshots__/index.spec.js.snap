// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`subject should error 1`] = `
Array [
  "test 1",
]
`;

exports[`subject should error 2`] = `
Array [
  [Error: error 1],
]
`;

exports[`subject should error 3`] = `
Array [
  "test 3",
]
`;

exports[`subject should ignore double unsub 1`] = `
Array [
  "test 1",
]
`;

exports[`subject should sub 1`] = `
Array [
  "test 1",
]
`;

exports[`subject should sub 2`] = `
Array [
  "test 2",
]
`;

exports[`subject should sub 3`] = `
Array [
  "test 3",
]
`;

exports[`subject should unsub 1`] = `
Array [
  "test 1",
]
`;
